tasks/extract-binaries-from-image/tests/test-extract-binaries-from-image.yaml

```diff
@@ -0,0 +1,106 @@
+---
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: test-extract-binaries-from-image
+spec:
+  description: |
+    Run the extract-binaries-from-image task
+  workspaces:
+    - name: tests-workspace
+  tasks:
+    - name: setup
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      taskSpec:
+        workspaces:
+          - name: data
+        steps:
+          - name: setup-snapshot
+            image: quay.io/redhat-appstudio/release-service-utils:447ea0580a2cdd48b4091e1df86fab5c3f86d01c
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+
+              cat > $(workspaces.data.path)/snapshot.json << EOF
+              {
+                "application": "myapp",
+                "components": [
+                  {
+                    "name": "comp",
+                    "containerImage": "registry.io/image:tag"
+                  }
+                ]
+              }
+              EOF
+
+              mkdir -p $(workspaces.data.path)/image_data
+              cd $(workspaces.data.path)/image_data
+
+              cat > manifest.json <<EOF
+              {
+                "layers": [
+                  {"digest": "sha256:1111"},
+                  {"digest": "sha256:2222"}
+                ]
+              }
+              EOF
+
+              TAR_IN1=$(mktemp -d)
```

Comment:
I'm a little confused by how this through L59 is doing anything. `mktemp` will create a directory in `/tmp`. This, AFAIK, won't be in the workspace (I believe the default path to the workspace is `/data`). So nothing done here is being used in the task IIUC

Reply:
This code is from  @mmalina, so he should be able to explain it better. But from what I understood. He creates the two temp directory but the tar files are created in `$(workspaces.data.path)/image_data` since the shell hasn't moved from that directory since L39. Then, in the `mocks.sh` he is doing this command: `cp $(workspaces.data.path)/image_data/* $TMP_DIR/`

Reply:
yeah, the idea was to prepare the data somewhere and then in the mock skopeo simulate the effect a real skopeo call would have - produce files in the target directory. Another option would be move even the files generation to the mock function.